demo_nodes_cpp/src/logging/use_logger_service.cpp

```diff
@@ -0,0 +1,207 @@
+    if (rclcpp::spin_until_future_complete(shared_from_this(), result) !=
+      rclcpp::FutureReturnCode::SUCCESS)
+    {
+      return false;
+    }
+
+    auto ret_result = result.get();
+    level = ret_result->levels[0].level;
+    return true;
+  }
+
+private:
+  const std::string remote_node_name_;
+  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub_;
+  rclcpp::Client<rcl_interfaces::srv::SetLoggerLevels>::SharedPtr logger_set_client_;
+  rclcpp::Client<rcl_interfaces::srv::GetLoggerLevels>::SharedPtr logger_get_client_;
+};
+
+
+int main(int argc, char ** argv)
+{
+  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
+  rclcpp::init(argc, argv);
+
+  const std::string node_name = "LoggerServiceNode";
+  auto logger_service_node = std::make_shared<LoggerServiceNode>(
+    node_name);
+  auto test_node = std::make_shared<TestNode>(node_name);
+
+  rclcpp::executors::SingleThreadedExecutor executor;
+
+  executor.add_node(logger_service_node);
+
+  std::thread thread([&executor]() {
+      executor.spin();
+    });
+
+  // Output with default logger level
+  RCLCPP_INFO(test_node->get_logger(), "Output with default logger level:");
+  {
+    auto msg = std::make_unique<std_msgs::msg::String>();
+    msg->data = "Output 1";
+    test_node->get_pub()->publish(std::move(msg));
+  }
+  std::this_thread::sleep_for(200ms);
+
+  // Output with debug logger level
+  RCLCPP_INFO(test_node->get_logger(), "Output with debug logger level:");
+  if (test_node->set_logger_level_on_remote_node(rclcpp::Logger::Level::Debug)) {
+    auto msg = std::make_unique<std_msgs::msg::String>();
+    msg->data = "Output 2";
+    test_node->get_pub()->publish(std::move(msg));
+    std::this_thread::sleep_for(200ms);
+  } else {
+    RCLCPP_ERROR(test_node->get_logger(), "Failed to set debug logger level via logger service !");
+  }
+
+
+  // Output with warn logger level
+  RCLCPP_INFO(test_node->get_logger(), "Output with warn logger level:");
+  if (test_node->set_logger_level_on_remote_node(rclcpp::Logger::Level::Warn)) {
+    auto msg = std::make_unique<std_msgs::msg::String>();
+    msg->data = "Output 3";
+    test_node->get_pub()->publish(std::move(msg));
+    std::this_thread::sleep_for(200ms);
+  } else {
+    RCLCPP_ERROR(test_node->get_logger(), "Failed to set debug logger level via logger service !");
+  }
+
+  // Output with error logger level
+  RCLCPP_INFO(test_node->get_logger(), "Output with error logger level:");
+  if (test_node->set_logger_level_on_remote_node(rclcpp::Logger::Level::Error)) {
+    auto msg = std::make_unique<std_msgs::msg::String>();
+    msg->data = "Output 4";
+    test_node->get_pub()->publish(std::move(msg));
+    std::this_thread::sleep_for(200ms);
+  } else {
+    RCLCPP_ERROR(test_node->get_logger(), "Failed to set debug logger level via logger service !");
+  }
+
+  // Should get error logger level (40)
+  uint8_t get_logger_level = 0;
+  if (test_node->get_logger_level_on_remote_node(get_logger_level)) {
```

Comment:
this can be called for each iteration to change the logger level on remote node?

Reply:
Okay

Reply:
Yes.